pkg/transform/oauth_transform.go

```diff
@@ -150,8 +152,22 @@ func (e OAuthTransform) Extract() (Extraction, error) {
 
 // Validate confirms we have recieved good OAuth configuration data during Extract
 func (e OAuthExtraction) Validate() error {
-	logrus.Warn("Oauth Transform Validation Not Implmeneted")
-	return nil // Simulate fine
+	var err error
+
+	serializer := k8sjson.NewYAMLSerializer(k8sjson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for _, identityProvider := range e.IdentityProviders {
+		switch identityProvider.Kind {
+		case "GitHubIdentityProvider":
+			err = oauth.ValidateGithubProvider(serializer, identityProvider)
```

Comment:
I don't think `OAuthExtraction` should know about the specific Provider types. I feel like it should simply call:
```golang
for _, identityProvider := range e.IdentityProviders {
    oauth.Validate(serializer, identityProvider)
}
```

Then `Validate` should live along side `Translate` in `oauth.go`. 

https://github.com/fusor/cpma/blob/master/pkg/transform/oauth/oauth.go#L89-L164

This way `Validate` can look similar to `Translate` by knowing to look at the Kind and decide which way to validate the individual providers.

Reply:
I don't feel too strongly about this either way, but if we're going to have the extraction essentially delegate `Transform` to the `oauth` package, it makes sense to do the same for `Validate`.

Reply:
@eriknelson @jmrodri I initially placed `Validate(oauthProviders)` in `ouath.go` when started writing this, but then moved it to `ouath_transform.go` because I wasn't sure where it should be. Now with your feedback, I know `ouath.go` is better, so I'm moving it back:)